SwEnt-Group8/Street-work-app
# Feat/Login state caching

### Summary

This Pull Request implements login caching, meaning that once the user is logged in with his google account, the application starts directly on the Map. This feature also implements the ability to see the user score and username on the application even when not connected to the Internet.

Implementing this function required firstly implementing a function that checks the application's connectivity to decide whether or not to load the user from the database or cache. It was also necessary to implement a simple Splash Screen to wait for the user to load before displaying the application.

This PR cherry picks commits from the soon to be closed PR https://github.com/SwEnt-Group8/Street-work-app/pull/166 due to features separation.

### Changes

- Add `NewtorkUtils.kt` file.
- Implement `isInternetAvailable` function in `NewtorkUtils.kt` file.
- Add a `SplashScreen.kt` file with a `SplashScreen` composable.
- Implement the login caching logic in `MainActivity.kt`.
- Implement the login caching logic in `SignIn.kt`.

### Tests

- Modify tests to implement the new feature signature and parameters.
- Add an UI test to test the `SplashScreen` composable.
- Add a `StreetWorkAppMainTest` file with tests.

**Please fully test this feature with your device / emulator to make sure it works properly.**
